src/index.js

```diff
@@ -1075,7 +1187,7 @@ const initializeFormElements = () => {
   watchNFTButton.onclick = async () => {
     let watchNftsResult;
     try {
-      watchNftsResult = await ethereum.request({
```

Comment:
in some places, we were using `ethereum.request` instead of provider. We need to make sure that we are using the provider (whichever is selected)